### PR TITLE
Split devices in family files

### PR DESCRIPTION
**Pull request by xanthio**
Contribution aiming to ease project maintenance:

1. We want to get rid off the large cube/cmsis/hal lists because:
    - they become larger when ST releases new families
    - concurrent  PR for updates create conflicts are they touch the same lines
2. I think minimizing the amount of file edited to introduction new family would encourage contributions

This major change was the opportunity to use the latest version of each ST repo.
But this needs 2 changes:

1. Better definition of shared memory block MB2 in generated `linker script` because WB startup initializes this memory area
2. Add a definition for MP1 family because clocks default values are set in HAL and CMSIS `system_<device>.c` doesn't include it if `USE_HAL_DRIVER` is not defined

### CODE REVIEW

**Comment by xanthio** [comment]
@Hish15: Here is the first maintenance PR

File: cmake/stm32/f0.cmake [hunk 1]
```diff
@@ -23,3 +23,89 @@ target_compile_options(STM32::F0 INTERFACE
...
+
+list(APPEND STM32_SUPPORTED_FAMILIES_LONG_NAME
+    STM32F0
+)
+
+list(APPEND STM32_FETCH_FAMILIES F0)
+
+set(CUBE_F0_VERSION  v1.11.5)
+set(CMSIS_F0_VERSION v2.3.7)
+set(HAL_F0_VERSION   v1.7.8)
```
Comment (Hish15) [inline cmake/stm32/f0.cmake:111]: Don't change anything on PR that moves so many things around
```suggestion
set(CUBE_F0_VERSION  v1.11.2)
set(CMSIS_F0_VERSION v2.3.5)
set(HAL_F0_VERSION   v1.7.5)

```

File: cmake/stm32/f1.cmake [hunk 2]
```diff
@@ -76,3 +76,111 @@ function(stm32f1_get_memory_info DEVICE TYPE FLASH_SIZE RAM_SIZE)
...
+
+list(APPEND STM32_SUPPORTED_FAMILIES_LONG_NAME
+    STM32F1
+)
+
+list(APPEND STM32_FETCH_FAMILIES F1)
+
+set(CUBE_F1_VERSION  v1.8.5)
+set(CMSIS_F1_VERSION v4.3.4)
+set(HAL_F1_VERSION   v1.1.9)
```
Comment (Hish15) [inline cmake/stm32/f1.cmake:186]:
```suggestion
set(CUBE_F1_VERSION  v1.8.4)
set(CMSIS_F1_VERSION v4.3.3)
set(HAL_F1_VERSION   v1.1.8)

```

File: cmake/stm32/f2.cmake [hunk 3]
```diff
@@ -37,3 +37,54 @@ function(stm32f2_get_memory_info DEVICE TYPE FLASH_SIZE RAM_SIZE)
...
+
+list(APPEND STM32_SUPPORTED_FAMILIES_LONG_NAME
+    STM32F2
+)
+
+list(APPEND STM32_FETCH_FAMILIES F2)
+
+set(CUBE_F2_VERSION  v1.9.4)
+set(CMSIS_F2_VERSION v2.2.6)
+set(HAL_F2_VERSION   v1.2.8)
```
Comment (Hish15) [inline cmake/stm32/f2.cmake:90]:
```suggestion
set(CUBE_F2_VERSION  v1.9.3)
set(CMSIS_F2_VERSION v2.2.5)
set(HAL_F2_VERSION   v1.2.7)
```

File: cmake/stm32/f3.cmake [hunk 4]
```diff
@@ -57,3 +57,85 @@ function(stm32f3_get_memory_info DEVICE TYPE FLASH_SIZE RAM_SIZE)
...
+
+list(APPEND STM32_SUPPORTED_FAMILIES_LONG_NAME
+    STM32F3
+)
+
+list(APPEND STM32_FETCH_FAMILIES F3)
+
+set(CUBE_F3_VERSION  v1.11.5)
+set(CMSIS_F3_VERSION v2.3.8)
+set(HAL_F3_VERSION   v1.5.8)
```
Comment (Hish15) [inline cmake/stm32/f3.cmake:141]:
```suggestion
set(CUBE_F3_VERSION  v1.11.2)
set(CMSIS_F3_VERSION v2.3.5)
set(HAL_F3_VERSION   v1.5.5)
```

File: cmake/stm32/g4.cmake [hunk 5]
```diff
@@ -23,3 +23,110 @@ target_compile_options(STM32::G4 INTERFACE
...
+
+list(APPEND STM32_SUPPORTED_FAMILIES_LONG_NAME
+    STM32G4
+)
+
+list(APPEND STM32_FETCH_FAMILIES G4)
+
+set(CUBE_G4_VERSION  v1.5.2)
+set(CMSIS_G4_VERSION v1.2.3)
+set(HAL_G4_VERSION   v1.2.3)
```
Comment (Hish15) [inline cmake/stm32/g4.cmake:132]:
```suggestion
set(CUBE_G4_VERSION  v1.4.0)
set(CMSIS_G4_VERSION v1.2.1)
set(HAL_G4_VERSION   v1.2.1)
```

**Review by Hish15 — CHANGES_REQUESTED** [review]
WIP : I checked that all devices are properly moved 👍🏿 
The CUBE/CMSIS/HAL must not be changed in this PR

**Review by Hish15 — APPROVED** [review]
@atsju please have a look and merge this, for me it's good to go.

**Pull request merged**